### ISSUE TITLE
[Bug] [Crash] Dynamic toolbar crash in `onWindowBoundsChanged`

### ISSUE THREAD

**Issue by csadilek** [mozilla-mobile/fenix#8973]
Saw this quite a few times in Sentry (with different values, of course):

```
java.lang.AssertionError: The maximum height of the dynamic toolbar (168) should be smaller than GeckoView height (158)
    at org.mozilla.geckoview.GeckoSession.onWindowBoundsChanged(GeckoSession.java:2)
    at org.mozilla.geckoview.GeckoSession.onSurfaceChanged(GeckoSession.java:8)
    at org.mozilla.geckoview.GeckoDisplay.surfaceChanged(GeckoDisplay.java:4)
    at org.mozilla.geckoview.GeckoDisplay.surfaceChanged(GeckoDisplay.java:1)
    at org.mozilla.geckoview.GeckoView$Display.onSurfaceChanged(GeckoView.java:2)
    at org.mozilla.gecko.SurfaceViewWrapper$ListenerWrapper.surfaceChanged(SurfaceViewWrapper.java:2)
    at android.view.SurfaceView.updateWindow(SurfaceView.java:668)
    at android.view.SurfaceView.setFrame(SurfaceView.java:340)
    at android.view.View.layout(View.java:18797)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
    at android.view.View.layout(View.java:18804)
    at android.view.ViewGroup.layout(ViewGroup.java:5951)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
    at android.view.View.layout(View.java:18804)
    at android.view.ViewGroup.layout(ViewGroup.java:5951)
    at androidx.swiperefreshlayout.widget.SwipeRefreshLayout.onLayout(SwipeRefreshLayout.java:11)
    at android.view.View.layout(View.java:18804)
    at android.view.ViewGroup.layout(ViewGroup.java:5951)
    at com.google.android.material.appbar.HeaderScrollingViewBehavior.layoutChild(HeaderScrollingViewBehavior.java:22)
    at com.google.android.material.appbar.ViewOffsetBehavior.onLayoutChild(ViewOffsetBehavior.java:1)
    at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:7)
    at android.view.View.layout(View.java:18804)
    at android.view.ViewGroup.layout(ViewGroup.java:5951)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
    at android.view.View.layout(View.java:18804)
    at android.view.ViewGroup.layout(ViewGroup.java:5951)
    at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1741)
    at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1585)
    at android.widget.LinearLayout.onLayout(LinearLayout.java:1494)
    at android.view.View.layout(View.java:18804)
    at android.view.ViewGroup.layout(ViewGroup.java:5951)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
    at android.view.View.layout(View.java:18804)
    at android.view.ViewGroup.layout(ViewGroup.java:5951)
    at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1741)
    at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1585)
    at android.widget.LinearLayout.onLayout(LinearLayout.java:1494)
    at android.view.View.layout(View.java:18804)
    at android.view.ViewGroup.layout(ViewGroup.java:5951)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
    at android.view.View.layout(View.java:18804)
    at android.view.ViewGroup.layout(ViewGroup.java:5951)
    at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1741)
    at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1585)
    at android.widget.LinearLayout.onLayout(LinearLayout.java:1494)
    at android.view.View.layout(View.java:18804)
    at android.view.ViewGroup.layout(ViewGroup.java:5951)
    at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
    at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
    at com.android.internal.policy.DecorView.onLayout(DecorView.java:891)
    at android.view.View.layout(View.java:18804)
    at android.view.ViewGroup.layout(ViewGroup.java:5951)
    at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2655)
    at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2371)
    at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1522)
    at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7098)
    at android.view.Choreographer$CallbackRecord.run(Choreographer.java:927)
    at android.view.Choreographer.doCallbacks(Choreographer.java:702)
    at android.view.Choreographer.doFrame(Choreographer.java:638)
    at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:913)
    at android.os.Handler.handleCallback(Handler.java:751)
    at android.os.Handler.dispatchMessage(Handler.java:95)
    at android.os.Looper.loop(Looper.java:154)
    at android.app.ActivityThread.main(ActivityThread.java:6682)
    at java.lang.reflect.Method.invoke(Method.java)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1534)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1424)
```

**Comment by softvision-oana-arbuzov** [comment]
I was able to reproduce the issue by following the steps:

**Steps to reproduce:**
1. Set 2 browsers in split view mode, one of them being Firefox Nightly Preview. 
2. Navigate to a site on both of the browsers (e.g nytimes.org)
3. On the browser, that is not Firefox Nightly Preview, perform an edit on the URL or a search.
4. Observe behavior.

**Expected result:**
The performed operation is made, and there is no impact on Firefox Nightly Preview browser.

**Actual result:**
Firefox Nightly Preview browser crashes and needs to be restarted.

**Environment:**
Browser / Version: Firefox Preview Nightly 200312 (🦎: 76.0a1-20200311041149)
Operating System: Huawei P20 Lite (Android 8.0.0) - 1080 x 2280 pixels, 19:9 ratio (~432 ppi density)

**Notes:**
1. The issue is not reproducible if instead of Firefox Preview Nightly, I use Firefox Beta 68.6b5.
2. Reproducible rate: 10/10
3. Gif attached.

![CrashSplitViewFF](https://user-images.githubusercontent.com/12184325/76633884-19078880-654e-11ea-9cf1-3ea2230ff779.gif)

**Comment by kbrosnan** [comment]
Fixed by bz-1621966

**Comment by softvision-oana-arbuzov** [comment]
I can confirm, I can no longer reproduce the crash.
![image](https://user-images.githubusercontent.com/12184325/89659895-3829fc80-d8d9-11ea-97f9-4e4b6b864afd.png)

**Tested with:**
Browser / Version: Firefox Nightly 200805 (🦎 81.0a1-20200803094100), Firefox Release 79.0.2 (🦎 79.0-20200720193547)
Operating System: Huawei P20 Lite (Android 8.0.0) - 1080 x 2280 pixels, 19:9 ratio (~432 ppi density)
